src/odemis/driver/xt_client.py

```diff
@@ -1756,10 +1760,12 @@ def _refreshPressure(self):
         self.position._set_value(val, force_write=True)
 
         # Pressure
-        pressure = self.parent.get_pressure()
-        self.pressure._set_value(pressure, force_write=True)
-
-        logging.debug("Updated chamber pressure, %s Pa, vacuum state %s.", pressure, val["vacuum"])
+        try:
+            pressure = self.parent.get_pressure()
+            self.pressure._set_value(pressure, force_write=True)
+            logging.debug("Updated chamber pressure, %s Pa, vacuum state %s.", pressure, val["vacuum"])
+        except IOError:
```

Comment:
I think it makes sense here to say:
```
except IOError:
    if pressure == -1:
        logging.warning("Couldn't read pressure value.")
    else:
        raise
```

Reply:
The way I implement it now is that in case of -1 pressure, an IOError is raised (and .get_pressure will not return anything) so it doesn't work. We could make a specific exception just for that case, but I think it's a bit over the top;. I don't think it's likely an IOError will be raised becaseu of other issues.